cluster-autoscaler/cloudprovider/gce/mig_info_provider.go

```diff
@@ -53,23 +54,27 @@ type MigInfoProvider interface {
 }
 
 type cachingMigInfoProvider struct {
-	migInfoMutex           sync.Mutex
-	cache                  *GceCache
-	migLister              MigLister
-	gceClient              AutoscalingGceClient
-	projectId              string
-	concurrentGceRefreshes int
-	migInstanceMutex       sync.Mutex
+	migInfoMutex                  sync.Mutex
+	cache                         *GceCache
+	migLister                     MigLister
+	gceClient                     AutoscalingGceClient
+	projectId                     string
+	concurrentGceRefreshes        int
+	migInstanceMutex              sync.Mutex
+	migInstancesMaxRefreshRate    time.Duration
+	migInstancesLastRefreshedInfo map[string]time.Time
 }
 
 // NewCachingMigInfoProvider creates an instance of caching MigInfoProvider
-func NewCachingMigInfoProvider(cache *GceCache, migLister MigLister, gceClient AutoscalingGceClient, projectId string, concurrentGceRefreshes int) MigInfoProvider {
+func NewCachingMigInfoProvider(cache *GceCache, migLister MigLister, gceClient AutoscalingGceClient, projectId string, concurrentGceRefreshes int, migInstancesMaxRefreshRate time.Duration) MigInfoProvider {
 	return &cachingMigInfoProvider{
-		cache:                  cache,
-		migLister:              migLister,
-		gceClient:              gceClient,
-		projectId:              projectId,
-		concurrentGceRefreshes: concurrentGceRefreshes,
+		cache:                         cache,
+		migLister:                     migLister,
+		gceClient:                     gceClient,
+		projectId:                     projectId,
+		concurrentGceRefreshes:        concurrentGceRefreshes,
+		migInstancesMaxRefreshRate:    migInstancesMaxRefreshRate,
```

Comment:
I would rename this variable (incl. in different places) to something on this lines of `migInstancesMinRefreshWaitTime`. 
We want to signify this is the amount of time you have to wait before refetching the mig instances.

Reply:
makes sense, changed